src/main/java/com/internship/backend/controller/ReservationController.java

```diff
@@ -28,10 +31,18 @@ public class ReservationController {
     private ReservationService reservationService;
 
     @PostMapping("/add")
-    public Reservation addReservation(@RequestBody ReservationDTO reservationDTO) throws ReservationAlreadyExists {
-        Reservation reservation = reservationService.fromDTO(reservationDTO);
-        reservationService.addReservation(reservation);
-        return new ResponseEntity<>(reservation, HttpStatus.CREATED).getBody();
+    public ResponseEntity<Reservation> addReservation(@RequestBody ReservationDTO reservationDTO) {
+        try{
+            Reservation reservation = reservationService.fromDTO(reservationDTO);
```

Comment:
Could the method name fromDTO be renamed into something like parse? If makes more sense.